src/lib/messaging/service.js

```diff
@@ -59,7 +57,6 @@ export default class Service extends BaseService {
     const promise = this.registerRequest(request, timeout)
 
     browser.tabs.sendMessage(tabID, request).catch(error => {
-      logger.error('Tabs.sendMessage() failed:', error)
```

Comment:
why remove these?

Reply:
Sometimes, when we reload a page with the extension active, a background script may send a message to the content script while the latter is not reloaded yet and is unavailable. As a result, the error message would be printed to the console. 

I was trying to eliminate this error message. The error in the situation described above is "legitimate" and should be of no concern to the user. Also, if request fails, we return a rejected promise from the send function. A code that is calling it can check for the promise value and decide whether a message should be printed or not. So instead of always printing a message here we can delegate that decision to the calling code that has more information about the context and can decide it better. Does it make sense? What do you think?

Reply:
yes, I agree. thanks for the explanation.